api/src/main/java/ai/djl/training/ParameterStore.java

```diff
@@ -119,14 +119,12 @@ public NDArray getValue(Parameter parameter, Device device, boolean training) {
                 }
             } else {
                 if (copy || !array.getDevice().equals(device)) {
-                    synchronized (array) {
-                        array = array.toDevice(device, true);
-                        array.attach(manager);
-                        // some parameter doesn't require grad
-                        // for example running_mean in BatchNorm
-                        if (parameter.requireGradient() && training) {
-                            array.attachGradient();
-                        }
+                    array = array.toDevice(device, true);
```

Comment:
We still need to evaluate this change. Can you ask them to test again if you would like to apply this fix?

Reply:
I am running testing against this change

Reply:
It turns out without adding the lock here, we still observe the SparseFormat error